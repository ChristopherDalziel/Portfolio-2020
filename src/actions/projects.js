import database from "../firebase/firebase";

export const createProject = (project) => ({
  type: "CREATE_PROJECT",
  project,
});

export const startCreateProject = (projectData = {}) => {
  return (dispatch) => {
    const {
      name = "",
      description = "",
      technology = "",
      githubUrl = "",
      url = "",
    } = projectData;
    const project = { name, description, technology, githubUrl, url };
    return database
      .ref(`projects`)
      .push(project)
      .then((ref) => {
        dispatch(createProject({ id: ref.key, ...project }));
      });
  };
};

export const setProjects = (projects) => ({
  type: "SET_PROJECTS",
  projects,
});

export const startSetProjects = () => {
  return (dispatch) => {
    return database
      .ref(`projects`)
      .once("value")
      .then((snapshot) => {
        const projects = [];

        snapshot.forEach((childSnapshot) => {
          projects.push({
            id: childSnapshot.key,
            ...childSnapshot.val(),
          });
        });

        dispatch(setProjects(projects));
      });
  };
};
