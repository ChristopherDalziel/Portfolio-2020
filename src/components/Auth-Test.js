import React from "react";

const Info = (props) => (
  <div>
    <h1>HOC or Higher Order Component</h1>
    <p>Info? {props.info}</p>
  </div>
);

const withAdminWarning = (WrappedComponent) => {
  return (props) => (
    <div>
      {props.isAdmin && <p>This is private info please don't share!</p>}
      {/* Spreading the props here creates a js expression and has the effect of taking every key value pair object and passes them down as props */}
      <WrappedComponent {...props} />
    </div>
  );
};

const requireAuthentication = (WrappedComponent) => {
  return (props) => (
    <div>
      {props.isAuthenticated ? (
        <WrappedComponent {...props} />
      ) : (
        "You're not authenticated to view this information"
      )}
    </div>
  );
};

export const AdminInfo = withAdminWarning(Info);
export const AuthInfo = requireAuthentication(Info);
