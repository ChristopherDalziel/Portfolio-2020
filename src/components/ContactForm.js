import React from "react";
import styled from "@emotion/styled";

const ContactContainer = styled.div`
  display: flex;
  flex-direction: column;
  background-color: transparent;
  height: 80%;
`;

const ContactContent = styled.div`
  margin: 200px 400px 0px 400px;
`;

const ContactInput = styled.input`
  width: 100%;
  border: none;
  background: transparent;
  margin: 0 0 5px;
  padding-top: 5px;
  padding-bottom: 5px;
  &:focus {
    outline: 0;
    border-bottom: 1px solid #ef626b;
  }
`;

const ContactTextArea = styled.textarea`
  border: none;
  height: 100px;
  width: 100%;
  max-width: 100%;
  resize: none;
  &:focus {
    outline: 0;
    border: 1px solid #ef626b;
  }
`;

const ContactButton = styled.button`
  cursor: pointer;
  width: 100%;
  border: none;
  background: #a8b7ab;
  color: #fff;
  margin: 0 0 5px;
  padding: 10px;
  font-size: 15px;
  &:hover {
    background: #ef626b;
    -webkit-transition: background 0.3s ease-in-out;
    -moz-transition: background 0.3s ease-in-out;
    transition: background-color 0.3s ease-in-out;
  }
`;

const ContactHeading = styled.h1`
  text-align: center;
  color: #493a37;
`;

class ContactForm extends React.Component {
  constructor(props) {
    super(props);
    this.submitForm = this.submitForm.bind(this);
    this.state = {
      status: "",
    };
  }

  render() {
    const { status } = this.state;
    return (
      <ContactContent>
        <ContactHeading>Contact Me.</ContactHeading>
        <form
          onSubmit={this.submitForm}
          action="https://formspree.io/xwkrbyzo"
          method="POST"
        >
          <label>Name:</label>
          <ContactInput type="name" name="name" />
          <label>Phone number:</label>
          <ContactInput type="number" name="number" />
          <label>Email:</label>
          <ContactInput type="email" name="email" />
          <label>Message:</label>
          <ContactTextArea type="text" name="message" />
          {status === "SUCCESS" ? (
            <p>Thankyou! Chris will be in contact with you soon.</p>
          ) : (
            <ContactButton>Submit</ContactButton>
          )}
          {status === "ERROR" && (
            <p>An error has occured, your email has not been submitted.</p>
          )}
        </form>
      </ContactContent>
    );
  }

  submitForm(ev) {
    ev.preventDefault();
    const form = ev.target;
    const data = new FormData(form);
    const xhr = new XMLHttpRequest();
    xhr.open(form.method, form.action);
    xhr.setRequestHeader("Accept", "application/json");
    xhr.onreadystatechange = () => {
      if (xhr.readyState !== XMLHttpRequest.DONE) return;
      if (xhr.status === 200) {
        form.reset();
        this.setState({ status: "SUCCESS" });
      } else {
        this.setState({ status: "ERROR" });
      }
    };
    xhr.send(data);
  }
}

export default ContactForm;
