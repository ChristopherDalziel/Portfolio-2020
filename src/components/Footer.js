import React from "react";
import styled from "@emotion/styled";

import {
  faTwitterSquare,
  faLinkedin,
  faGithubSquare,
} from "@fortawesome/free-brands-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

const Footer = styled.footer`
  display: flex;
  flex-direction: row;
  position: absolute;
  bottom: 0;
  max-width: 100vw;
  height: 10%;
  width: 100%;
  padding-bottom: 8px;
  @media (max-width: 1024px) {
    background-color: white;
    height: 4%;
  }
  @media (max-width: 950px) {
    background-color: transparent;
  }
`;

const FooterLinks = styled.a`
  display: flex;
  align-items: flex-end;
  margin: 5px;
  color: #493a37;
  font-size: 50px;
  &:hover {
    color: #a8b7ab;
    -webkit-transition: color 0.3s ease-in-out;
    -moz-transition: color 0.3s ease-in-out;
    transition: color 0.3s ease-in-out;
  }
  @media (max-width: 1024px) {
    margin: 0;
    margin-left: 8px;
    font-size: 34px;
  }
`;

const FooterComponent = () => {
  return (
    <Footer>
      <FooterLinks target="_blank" href="https://twitter.com/dalziel_chris">
        <FontAwesomeIcon icon={faTwitterSquare} />
      </FooterLinks>
      <FooterLinks
        target="_blank"
        href="https://www.linkedin.com/in/christopher-dalziel-129427187/"
      >
        <FontAwesomeIcon icon={faLinkedin} />
      </FooterLinks>
      <FooterLinks target="_blank" href="https://github.com/ChristopherDalziel">
        <FontAwesomeIcon icon={faGithubSquare} />
      </FooterLinks>
    </Footer>
  );
};

export default FooterComponent;
