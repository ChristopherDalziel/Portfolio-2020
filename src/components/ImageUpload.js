// import React, { useState } from "react";
// import { storage } from "../firebase/firebase";

// const ImageUpload = () => {
//   const [image, setImage] = useState(null);
//   const [url, setUrl] = useState("");

//   const onImageChange = (e) => {
//     if (e.target.files[0]) {
//       setImage(e.target.files[0]);
//     }
//   };

//   const handleImageUpload = (e) => {
//     e.preventDefault();
//     const uploadTask = storage.ref(`images/${image.name}`).put(image);
//     uploadTask.on(
//       "state_changed",
//       (snapshot) => {},
//       (error) => {
//         console.log(error);
//       },
//       () => {
//         storage
//           .ref("images")
//           .child(image.name)
//           .getDownloadURL()
//           .then((url) => {
//             setUrl(url);
//           });
//       }
//     );
//   };

//   return (
//     <>
//       <input type="file" onChange={onImageChange} />
//       <button onClick={handleImageUpload}>Upload Image</button>
//     </>
//   );
// };

// export default ImageUpload;
