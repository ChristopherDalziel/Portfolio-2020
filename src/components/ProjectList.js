import React from "react";
import { connect } from "react-redux";
import Project from "./Project";

const ProjectList = (props) => (
  <>
    <Project />
    {props.projects.map((project) => {
      return <Project key={project.id} {...project} />;
    })}
  </>
);

const mapStateToProps = (state) => {
  return {
    projects: state.projects,
  };
};

export default connect(mapStateToProps)(ProjectList);
