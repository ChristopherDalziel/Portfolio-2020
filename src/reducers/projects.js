const projectReducerDefaultState = [];

const projectReducer = (state = projectReducerDefaultState, action) => {
  switch (action.type) {
    case "CREATE_PROJECT":
      return [...state, action.project];
    case "SET_PROJECTS":
      return action.projects;
    default:
      return state;
  }
};

export default projectReducer;
