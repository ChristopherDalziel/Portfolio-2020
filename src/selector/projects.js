const getVisibleProjects = (projects, { text }) => {
  return projects.filter((project) => {
    const matchText =
      project.name.toLowerCase().includes(text.toLowerCase()) ||
      project.technology.toLowerCase().includes(text.toLowerCase());

    return matchText;
  });
};

export default getVisibleProjects;
