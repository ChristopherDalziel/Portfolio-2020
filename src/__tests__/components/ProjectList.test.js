import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import { ProjectList } from "../../components/ProjectList";
import projects from "../fixtures/projects";

test("Should render list with projects", () => {
  const wrapper = shallow(<ProjectList projects={projects} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

test("Should render project list with empty message", () => {
  const wrapper = shallow(<ProjectList projects={[]} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});
