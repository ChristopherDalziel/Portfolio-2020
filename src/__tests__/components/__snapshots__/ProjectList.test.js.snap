// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render list with projects 1`] = `
<Styled(div)>
  <Project
    description="Fixture Project 1"
    githubUrl="http//fakegithuburl.com"
    id="1"
    key="1"
    name="Fixture Project 1"
    technology="Ruby"
    url="Image Url"
  />
  <Project
    description="Fixture Project 2"
    githubUrl="http//fakegithuburl.com"
    id="2"
    key="2"
    name="Fixture Project 2"
    technology="JavaScript"
    url="Image Url"
  />
  <Project
    description="Fixture Project 3"
    githubUrl="http//fakegithuburl.com"
    id="3"
    key="3"
    name="Fixture Project 3"
    technology="Elixir"
    url="Image Url"
  />
</Styled(div)>
`;

exports[`Should render project list with empty message 1`] = `
<Styled(div)>
  <p>
    No project(s) to display right now
  </p>
</Styled(div)>
`;
